### PR TITLE
network: add CAN Termination tristate option

### PR DESCRIPTION
**Pull request by mranostay**

### CODE REVIEW

File: src/network/networkd-can.c [hunk 1]
```diff
@@ -152,6 +152,22 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_CTRLMODE attribute: %m");
         }
 
+        if (link->network->can_termination >= 0) {
+                uint64_t termination = link->network->can_termination;
+
+                if (termination > UINT16_MAX) {
+                        log_link_error(link, "invalid can-termination");
+                        return -ERANGE;
+                }
```
Comment (yuwata) [inline src/network/networkd-can.c]: Please do not check the value range here, but introduce, say `config_parse_can_termination()`, and check the value inside the function.

File: src/network/networkd-can.c [hunk 2]
```diff
@@ -152,6 +152,22 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_CTRLMODE attribute: %m");
         }
 
+        if (link->network->can_termination >= 0) {
+                uint64_t termination = link->network->can_termination;
```
Comment (yuwata) [inline src/network/networkd-can.c]: This assignment is not necessary.

File: src/network/networkd-can.c [hunk 3]
```diff
@@ -152,6 +154,17 @@ static int link_set_can(Link *link) {
                         return log_link_error_errno(link, r, "Could not append IFLA_CAN_CTRLMODE attribute: %m");
         }
 
+        if (link->network->can_termination >= 0) {
+
+                log_link_debug(link, "%sabling can-termination", link->network->can_termination ? "En" : "Dis");
+
+                r = sd_netlink_message_append_u16(m, IFLA_CAN_TERMINATION,
+                               link->network->can_termination ? CAN_TERMINATION_OHM_VALUE : 0);
```
Comment (yuwata) [inline src/network/networkd-can.c]: indentation?

Comment (mranostay) [reply src/network/networkd-can.c]: As this should be on one line?

Comment (yuwata) [reply src/network/networkd-can.c]: 8ch indentation please.

**Comment by mranostay** [comment]
@yuwata Somehow the CI bot is still saying "Changed Needed" but have been resolved in last version.

**Comment by yuwata** [comment]
LGTM.

**Comment by yuwata** [comment]
Replaced by #15036.

**Pull request closed without merging**